### ISSUE TITLE
Firmware not uploading

### ISSUE THREAD

**Issue by streakerau** [Dealerpriest/me-high-power-motorboard-firmware#2]
Hi guys, thanks for sharing this, the noise is exactly the problem we are trying to solve with our robot.

I'm trying to upload your firmware to the encoder board using and FDTI board. I connected 5c gnd tx and rx as suggested and selected Arduino Uno as board in the Arduino software.

However it's returning an error during the upload phase:

avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 1 of 10: not in sync: resp=0x00
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 2 of 10: not in sync: resp=0x00

Can you please advice which board you select from the arduino IDE?
thanks

**Comment by gunhaxxor** [comment]
I'm not sure, but to me it sounds like the FTDI converter is found and that you have specified the correct serial port. But that the actual upload over over the rx tx pins fail. Have you connected the rx of the FTDI to the tx of the encoder board ( and the tx to rx)? Have you disconnected any rj25 cables before upload?

Also. Try with different usb-cable and usb port. And with different jumper wires.

Regarding board. Yes arduino uno is the correct one (called arduino/genuino uno in the later versions of the arduino software i think).
You could try to upload a more basic sketch, like the blink example, although I find it unlikely that it should make a difference.

I also want to take this opportunity to clarify that this firmware doesn't implement all the functionality in the original firmware, but only the parts that are relevant to us. Which is basically setting the speed of the motor :-)

**Comment by streakerau** [comment]
Thanks for the quick reply.
I managed to upload the firmware just pressing the reset button on the board before starting the upload from the Arduino IDE.

We also just need the motor speed control for our application. I have another question around how to set speed but I will add another issue. Closing this one. thanks
